### PR TITLE
Parse connection string by splitting on first =

### PR DESCRIPTION
**Pull request by archoversight**
Certain parameters may contain more than one = sign in the string, for
example the SharedAccessSignature.

This allows a connection string as follows:

    AccountName=guywald;SharedAccessSignature=se=2036-01-01&sp=acw&sv=2018-11-09&sr=c&sig=c2lnbmF0dXJlCg%3D%3D

To be parsed correctly into its component pieces:

    AccountName: guywald
    SharedAccessSignature: AccountName=guywald;SharedAccessSignature=se=2036-01-01&sp=acw&sv=2018-11-09&sr=c&sig=c2lnbmF0dXJlCg%3D%3D

Previously it would get truncated as:

    SharedAccessSignature: se

### CODE REVIEW

**Comment by archoversight** [comment]
The Azure Portal when you create an SAS will create the following connection string:

```
BlobEndpoint=https://logs.blob.core.usgovcloudapi.net/;QueueEndpoint=https://logs.queue.core.usgovcloudapi.net/;FileEndpoint=https://logs.file.core.usgovcloudapi.net/;TableEndpoint=https://logs.table.core.usgovcloudapi.net/;SharedAccessSignature=sv=2020-08-04&ss=bfqt&srt=sco&sp=rwdlacupx&se=2022-01-22T07:47:17Z&st=2022-01-21T23:47:17Z&spr=https&sig=c2lnbmF0dXJlCg%3D%3D
```

**Review by cataggar — APPROVED** [review]
Thanks!

File: sdk/storage/src/core/connection_string.rs [hunk 1]
```diff
@@ -104,26 +104,23 @@ impl<'a> ConnectionString<'a> {
...
+            let (k, v) = match kv {
+                Some((k, _)) if k.chars().all(char::is_whitespace) => {
                     return Err(ConnectionStringError::ParseError {
                         msg: "No key found".to_owned(),
                     })
                 }
+                Some((k, v)) if v.chars().all(char::is_whitespace) => {
+                    return Err(ConnectionStringError::MissingValue { key: k.to_owned() })
+                }
+                Some((k, v)) => (k, v),
```
[the remark this answers is not in the document]
Comment (archoversight) [reply sdk/storage/src/core/connection_string.rs]: I've fixed this now, and added additional testing so that this can't regress.

**Comment by archoversight** [comment]
Re-pushed after formatting.

**Review by ctaggart — APPROVED** [review]
All the new test are appreciated.

**Comment by archoversight** [comment]
Of course, helped me validate that the fix I was writing was actually correct too! Thanks for merging!

**Comment by cataggar** [comment]
Sorry, this isn't in the 0.1.0 that I just published. I was on the previous commit.

**Comment by archoversight** [comment]
@cataggar that's okay for right now, vector the tool that I am using to send logs to Azure blob storage is still using the git repository directly:

https://github.com/vectordotdev/vector/pull/11030

A future update will hopefully start using the crates instead of pulling from git, but I wasn't sure what your release time line looked like.

**Pull request merged**